conda_smithy/update_cb3.py

```diff
@@ -0,0 +1,386 @@
+    __slots__ = ('lc')
+
+    style = ""
+
+    def __new__(cls, value):
+        return ruamel.yaml.scalarstring.ScalarString.__new__(cls, value)
+
+class MyPreservedScalarString(ruamel.yaml.scalarstring.PreservedScalarString):
+    __slots__ = ('lc')
+
+class MyDoubleQuotedScalarString(ruamel.yaml.scalarstring.DoubleQuotedScalarString):
+    __slots__ = ('lc')
+
+class MySingleQuotedScalarString(ruamel.yaml.scalarstring.SingleQuotedScalarString):
+    __slots__ = ('lc')
+
+class MyConstructor(ruamel.yaml.constructor.RoundTripConstructor):
+    def construct_scalar(self, node):
+        # type: (Any) -> Any
+        if not isinstance(node, ruamel.yaml.nodes.ScalarNode):
+            raise ruamel.yaml.constructor.ConstructorError(
+                None, None,
+                "expected a scalar node, but found %s" % node.id,
+                node.start_mark)
+
+        if node.style == '|' and isinstance(node.value, ruamel.yaml.compat.text_type):
+            ret_val = MyPreservedScalarString(node.value)
+        elif bool(self._preserve_quotes) and isinstance(node.value, ruamel.yaml.compat.text_type):
+            if node.style == "'":
+                ret_val = MySingleQuotedScalarString(node.value)
+            elif node.style == '"':
+                ret_val = MyDoubleQuotedScalarString(node.value)
+            else:
+                ret_val = Str(node.value)
+        else:
+            ret_val = Str(node.value)
+        ret_val.lc = ruamel.yaml.comments.LineCol()
+        ret_val.lc.line = node.start_mark.line
+        ret_val.lc.col = node.start_mark.column
+        return ret_val
+
+
+class Section:
+    def __init__(self, section, start, end):
+        self.section = section
+        self.start = start
+        self.end = end
+
+    def __getitem__(self, item):
+        if item not in self.section:
+            return None
+        sect = self.section[item]
+        start = sect.lc.line
+        for other in self.section:
+            if other.lc.line > start:
+                end = other.lc.line
+                return Section(sect, start, end)
+        return Section(sect, start, self.end)
+
+
+class NullUndefined(jinja2.Undefined):
+    def __unicode__(self):
+        return self._undefined_name
+
+    def __getattr__(self, name):
+        return '{}.{}'.format(self, name)
+
+    def __getitem__(self, name):
+        return '{}["{}"]'.format(self, name)
+
+
+def get_compilers(url):
+    '''
+    Download the source and check for C/C++/Fortran
+    Also check if `np.get_include()` is present in the setup.py files
+    Return whether a C/C++/Fortran compiler is used and whether
+    numpy headers are used.
+    '''
+    if isinstance(url, list):
+        for u in url:
+            r = requests.get(u, allow_redirects=True)
+            if r.ok:
+                url = u
+                break
+    else:
+        r = requests.get(url, allow_redirects=True)
+    fname = url.split('/')[-1]
+    tmp_dir = tempfile.mkdtemp('modernize')
```

Comment:
We might need to delete this file as storage is limited in heroku and travis

Reply:
I mean after we are done with the folder

Reply:
can this use the context manager syntax?